xtable-core/src/main/java/org/apache/xtable/parquet/ParquetConversionSource.java

```diff
@@ -0,0 +1,264 @@
+                                                        basePath))
+                                        .lastModified(file.getModificationTime())
+                                        .columnStats(
+                                                parquetStatsExtractor.getColumnStatsForaFile(
+                                                        parquetMetadataExtractor.readParquetMetadata(
+                                                                hadoopConf, file.getPath())))
+                                        .build())
+                .collect(Collectors.toList());
+    }
+
+    private InternalDataFile createInternalDataFileFromParquetFile(FileStatus parquetFile) {
+        return InternalDataFile.builder()
+                .physicalPath(parquetFile.getPath().toString())
+                .partitionValues(
+                        partitionValueExtractor.extractPartitionValues(
+                                partitionValueExtractor.extractParquetPartitions(
+                                        parquetMetadataExtractor.readParquetMetadata(hadoopConf, parquetFile.getPath()),
+                                        parquetFile.getPath().toString()),
+                                basePath))
+                .lastModified(parquetFile.getModificationTime())
+                .fileSizeBytes(parquetFile.getLen())
+                .columnStats(
+                        parquetStatsExtractor.getColumnStatsForaFile(
+                                parquetMetadataExtractor.readParquetMetadata(hadoopConf, parquetFile.getPath())))
+                .build();
+    }
+
+    @Override
+    public CommitsBacklog<Long> getCommitsBacklog(InstantsForIncrementalSync syncInstants) {
+        // based on either table formats?
+        List<Long> commitsToProcess =
+                Collections.singletonList(syncInstants.getLastSyncInstant().toEpochMilli());
+        return CommitsBacklog.<Long>builder().commitsToProcess(commitsToProcess).build();
+    }
+
+    @Override
+    public TableChange getTableChangeForCommit(Long modificationTime) {
+        List<LocatedFileStatus> parquetFiles = getParquetFiles(hadoopConf, basePath);
+        Set<InternalDataFile> addedInternalDataFiles = new HashSet<>();
+
+        List<FileStatus> tableChangesAfter =
+                parquetFiles.stream()
+                        .filter(fileStatus -> fileStatus.getModificationTime() > modificationTime)
+                        .collect(Collectors.toList());
+        List<FileStatus> tableChangesBefore =
+                parquetFiles.stream()
+                        .filter(fileStatus -> fileStatus.getModificationTime() < modificationTime)
+                        .collect(Collectors.toList());
+        InternalTable internalTable = getMostRecentTable(parquetFiles);
+
+        for (FileStatus tableStatus : tableChangesAfter) {
+            boolean isPresent = tableChangesBefore.contains(tableStatus);
```

Comment:
I think we can skip this step as the sets of FileStatus should be distinct due to the condition on the modification time